test/integration/protocol_integration_test.cc

```diff
@@ -1321,9 +1319,14 @@ TEST_P(ProtocolIntegrationTest, MissingStatus) {
     ASSERT(fake_upstream_connection->write(std::string(ack_frame)));     // ack setting
     Http::Http2::Http2Frame missing_status = Http::Http2::Http2Frame::makeHeadersFrameNoStatus(0);
     ASSERT_TRUE(fake_upstream_connection->write(std::string(missing_status)));
+    ASSERT_TRUE(fake_upstream_connection->waitForDisconnect());
+  } else {
+    waitForNextUpstreamRequest();
+    default_response_headers_.removeStatus();
+    upstream_request_->encodeHeaders(default_response_headers_, false);
+    ASSERT_TRUE(upstream_request_->waitForReset());
   }
 
-  ASSERT_TRUE(fake_upstream_connection->waitForDisconnect());
```

Comment:
I suspect watiForDisconnect() here and above will be fixed by https://github.com/envoyproxy/envoy/pull/15753. Mind adding a TODO here?